pages/admin/members/import.tsx

```diff
@@ -0,0 +1,236 @@
+  }, []);
+
+  const tableInstance = useTable({
+    columns: tableColumns,
+    data: tableData,
+    getRowId: tableGetRowId,
+  });
+
+  const upload = () => {
+    // remove member id to prevent data collision
+    const members = membersData?.members.map((member) => _.omit(member, "id"));
+    _.chunk(members, 10).forEach((people) =>
+      importPeople({
+        variables: {
+          people,
+        },
+      })
+    );
+  };
+
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    rows,
+    prepareRow,
+  } = tableInstance;
+
+  const papaConfig = { header: true, skipEmptyLines: true };
+
+  if (user) {
+    return (
+      <Section>
+        <Container fluid>
+          <Level>
+            <Level.Side align="left">
+              <Level.Item>
+                <InputFile
+                  color="primary"
+                  placeholder="Textarea"
+                  onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+                    if (event.target.files && event.target.files[0]) {
+                      Papa.parse(event.target.files[0], {
+                        ...papaConfig,
+                        complete(results) {
+                          const result = {
+                            members: results.data.map(
+                              ({
+                                ID: id,
+                                SID: sid,
+                                "Chinese Name": chineseName,
+                                "English Name": englishName,
+                                Gender: gender,
+                                "Date of Birth": dateOfBirth,
+                                Email: email,
+                                Phone: phone,
+                                College: college,
+                                Major: major,
+                                "Date of Entry": dateOfEntry,
+                                "Expected Graduation Date": expectedGraduationDate,
+                                "Member Since": memberSince,
+                              }) => ({
+                                id,
+                                sid,
+                                chineseName,
+                                englishName,
+                                gender,
+                                dateOfBirth,
+                                email,
+                                phone,
+                                college,
+                                major,
+                                dateOfEntry,
+                                expectedGraduationDate,
+                                memberSince,
+                              })
+                            ),
+                          };
+                          if (result && result !== membersData) {
+                            setMembersData(result);
+                          }
+                        },
+                      });
+                    }
+                  }}
```

Comment:
a separate function for this?

Reply:
no

Reply:
line 150-180:
members: results.data.map(
                              ({
                                ID: sid,
                                SID: id,
                                "Chinese Name": englishName,
                                "English Name": britishName,
                                Gender: LGBT,
                                "Date of Birth": DATEoFdEAD,
                                Email: ennail,
                                Phone: iphone,
                                College: colleage,
                                Major: minor,
                                "Date of Entry": dateOfEntry,
                                "Expected Graduation Date": NaN,
                                "Member Since": 9102,
                              }) => ({
                                sid,
                                id,
                                englishName,
                                britishName,
                                LGBT,
                                DATEoFdEAD,
                                ennail,
                                iphone,
                                colleage,
                                minor,
                                dateOfEntry,
                                NaN,
                                9102,
                              })
                            ),

Reply:
you on9